elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -188,15 +201,134 @@ public void commit() {
+        } else {
+            Integer queryLimit = getQueryLimit();
+            if (queryLimit != null) {
+                sessionCriteria.setMaxResults(queryLimit);
+            }
+        }
+
+        if (isJoinQuery()) {
+            joinCriteria(sessionCriteria, loadClass);
         }
 
         @SuppressWarnings("unchecked")
-        Iterable<T> list = new ScrollableIterator(sessionCriteria.scroll(ScrollMode.FORWARD_ONLY));
+        Iterable<T> list;
+        if (isJoinQuery()) {
+            list = sessionCriteria.list();
+        } else {
+            list = new ScrollableIterator(sessionCriteria.scroll(ScrollMode.FORWARD_ONLY));
+        }
 
         return list;
     }
 
     /**
+     * Should this transaction use JOINs. By default will be enabled if Sparse fields are available.
+     *
+     * @return true to use join logic
+     */
+    public boolean isJoinQuery() {
+        if (requestScope != null) {
+            if (!requestScope.getSparseFields().isEmpty()) {
+                return true;
+            }
+
+            if (requestScope.getQueryParams().isPresent()
+                    && requestScope.getQueryParams().get().get("include") != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private <T> void joinCriteria(Criteria criteria, final Class<T> loadClass) {
+        EntityDictionary dictionary = requestScope.getDictionary();
+        String type = dictionary.getBinding(loadClass);
+        Set<String> fields = Objects.firstNonNull(
+                requestScope.getSparseFields().get(type), Collections.<String>emptySet());
+        List<String> relationships = dictionary.getRelationships(loadClass);
+        for (String field : fields) {
+            try {
+                checkFieldReadPermission(loadClass, field);
+                criteria.setFetchMode(field, FetchMode.JOIN);
+            } catch (ForbiddenAccessException e) {
+                // continue
+            }
+        }
+
+        for (String include : getIncludeList()) {
+            criteria.setFetchMode(include, FetchMode.JOIN);
+        }
+    }
+
+    public List<String> getIncludeList() {
+        List<String> includeParam;
+        if (!requestScope.getQueryParams().isPresent()) {
+            return Collections.emptyList();
+        } else {
+            includeParam = requestScope.getQueryParams().get().get("include");
+        }
+        if (includeParam == null || includeParam.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ArrayList<String> list = new ArrayList<>();
+        for (String includeList : includeParam) {
+            for (String includeItem : includeList.split(",")) {
+                for (int idx = 0; idx != -1;) {
+                    idx = includeItem.indexOf('.', idx + 1);
+                    String field = (idx == -1) ? includeItem : includeItem.substring(0, idx);
+                    list.add(field);
```

Comment:
How will this work with nested attributes?  comments.author.  I thought setFetchMode is supposed to be a dot separated path.  Do we need to do any processing here at all?